### PR TITLE
Updates warning with instructions to rebuild usb drivers without reinstalling t2-cli

### PR DESCRIPTION
**Pull request by HipsterBrown**
This PR proposes a new solution to the "WARNING: Please run npm install -g t2-cli to compile USB drivers for the newer version of node" issue. (More details [here](https://github.com/tessel/t2-cli/issues/1071))

The error message will now show: **"ERR! Node version mismatch for USB drivers. Please run "npm rebuild --update-binary usb" to rebuild the USB drivers for your version of node"**

<del> `t2 rebuild` is new command that programmatically calls `npm rebuild usb --update-binary` so users don't have to completely reinstall the CLI. </del>

Edit: After a conversation with @rwaldron, I learned that `npm rebuild --update-binary usb` can be called from anywhere. So instead of adding this responsibility to this CLI, we can simply inform the user to call the `npm rebuild` command manually. 

Preview:

```
hipsterbrown:tessel-test $ t2 list
ERR! Node version mismatch for USB drivers.
INFO To correct this issue, please run the following command:
INFO
INFO   npm rebuild --update-binary usb
INFO
INFO This will rebuild the USB drivers for your version of Node.js
hipsterbrown:tessel-test $ npm rebuild --update-binary usb

> usb@1.2.0 install /Users/hipsterbrown/Projects/tessel-test/node_modules/t2-cli/node_modules/usb
> node-pre-gyp install --fallback-to-build

[usb] Success: "/Users/hipsterbrown/Projects/t2-cli/node_modules/usb/src/binding/usb_bindings.node" is installed via remote
usb@1.2.0 /Users/hipsterbrown/Projects/tessel-test/node_modules/t2-cli/node_modules/usb
hipsterbrown:tessel-test $ t2 list
INFO Searching for nearby Tessels...
WARN No Tessels Found.
hipsterbrown:tessel-test $
```

### CODE REVIEW

**Comment by Frijol** [comment]
Cool! I'd suggest a pretty explicit command name like "usb-driver-rebuild" since it's used infrequently and otherwise might be confused with rebuilding a tessel project or similar

**Comment by HipsterBrown** [comment]
@Frijol I've updated the description with a new direction for this PR.

File: lib/usb-connection.js [hunk 1]
```diff
@@ -22,14 +22,19 @@ function debugCommands(message) {
...
   // var VENDOR_REQ_IN  = usb.LIBUSB_REQUEST_TYPE_VENDOR | usb.LIBUSB_RECIPIENT_DEVICE | usb.LIBUSB_ENDPOINT_IN;
-} catch (e) {
-  haveusb = false;
-  log.error('WARNING: No USB controller found on this system. Please run npm install -g t2-cli to compile USB drivers for your version of node');
+} catch (error) {
+  isUSBAvailable = false;
+
+  // do not exit the process during tests because usb is not needed to run them
+  if (!global.IS_TEST_ENV) {
+    log.error('Node version mismatch for USB drivers. Please run "npm rebuild --update-binary usb" to rebuild the USB drivers for your version of node');
```
Comment (rwaldron) [inline lib/usb-connection.js]: Let's take this opportunity to improve this for our users. Something like this: 

```js
const tags = require('common-tags');
```

```js
log.error('Node version mismatch for USB drivers.');
log.info(tags.stripIndent `
  To correct this issue, run the following command: 

  npm rebuild --update-binary usb

  This will rebuild the USB drivers for your version of Node.js
`);
```

File: lib/controller.js [hunk 2]
```diff
@@ -181,6 +182,8 @@ Tessel.list = function(opts) {
 };
 
 Tessel.get = function(options) {
+  var discover = require('./discover');
+
```
Comment (rwaldron) [inline lib/controller.js]: Are these changes still necessary?

File: lib/init/javascript.js [hunk 3]
```diff
@@ -21,10 +21,11 @@ exportables.meta = {
   keywords: ['javascript', 'js']
 };
 
-exportables.loadNpm = () => {
+exportables.loadNpm = (options) => {
+  options = options || {};
   // You have to load npm in order to use it programatically
   return new Promise((resolve, reject) => {
-    NPM.load((error, npm) => {
+    NPM.load(options, (error, npm) => {
```
Comment (rwaldron) [inline lib/init/javascript.js]: Are these changes still necessary?

Comment (HipsterBrown) [inline lib/init/javascript.js]: This part is no longer necessary. I can revert those changes.

File: lib/tessel/deployment/javascript.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ var Tessel = require('../tessel');
 var binaryModulesUsed = new Map();
 
 const BINARY_SERVER_ROOT = `http://${remote.PACKAGES_HOSTNAME}/npm/`;
-const BINARY_CACHE_PATH = path.join(Tessel.LOCAL_AUTH_PATH, 'binaries');
+const BINARY_CACHE_PATH = path.join(Tessel.LOCAL_AUTH_PATH || '', 'binaries');
```
Comment (rwaldron) [inline lib/tessel/deployment/javascript.js]: Are these changes still necessary?

File: test/unit/init.js [hunk 5]
```diff
@@ -444,7 +444,7 @@ exports['init --lang=javascript'] = {
     test.expect(2);
 
     this.npmLoad.restore();
-    this.npmLoad = this.sandbox.stub(npm, 'load', (callback) => {
+    this.npmLoad = this.sandbox.stub(npm, 'load', (options, callback) => {
```
Comment (rwaldron) [inline test/unit/init.js]: Are these changes still necessary?

File: lib/controller.js (hunk 2 above)
Comment (HipsterBrown) [inline lib/controller.js]: I believe so, yes. When the `discover` module is always required by this file, any CLI action can be blocked by the USB driver error, including things like `t2 init`. Requiring the `discover` only when needed, _should_ lead to a nicer user experience.

File: lib/tessel/deployment/javascript.js (hunk 4 above)
Comment (HipsterBrown) [inline lib/tessel/deployment/javascript.js]: I don't believe so. I'll revert those after I run the tests.

File: test/unit/init.js (hunk 5 above)
Comment (HipsterBrown) [inline test/unit/init.js]: Nope, I'll revert that.

File: lib/controller.js (hunk 2 above)
Comment (rwaldron) [inline lib/controller.js]: Ok, but I think we should revisit this some in the future

File: lib/controller.js [hunk 6]
```diff
@@ -12,14 +12,13 @@ var semver = require('semver');
...
 var CrashReporter = require('./crash-reporter');
-var discover = require('./discover');
 var init = require('./init');
 var installer = require('./installer');
 var log = require('./log');
 var Menu = require('./menu');
 var updates = require('./update-fetch');
-var provision = require('./tessel/provision');
 var Tessel = require('./tessel/tessel');
+var provision = require('./tessel/provision');
```
Comment (HipsterBrown) [inline lib/controller.js]: Reordered this so that certain constants, i.e. `Tessel.LOCAL_AUTH_PATH`, can be added correctly.

**Comment by HipsterBrown** [comment]
The description has been updated to preview the new formatted error and info logs.

File: lib/usb-connection.js [hunk 7]
```diff
@@ -22,14 +22,26 @@ function debugCommands(message) {
...
+    log.error('Node version mismatch for USB drivers.');
+    log.info(tags.stripIndent`
+      To correct this issue, please run the following command:
+
+        npm rebuild --update-binary usb
+
+      This will rebuild the USB drivers for your version of Node.js
+    `);
+    process.exit(1);
+  }
```
Comment (rwaldron) [inline lib/usb-connection.js]: 👍
